src/os/bluestore/BlueFS.cc

```diff
@@ -3022,3 +3161,268 @@ bool BlueFS::wal_is_rotational()
   }
   return bdev[BDEV_SLOW]->is_rotational();
 }
+
+#undef dout_prefix
+#define dout_prefix *_dout << "bluefs rebalance "
+
+void BlueFS::consider_move()
+{
+  //this function does make sense only if both BDEV_DB and BDEV_SLOW are used
+  /*
+   * There are 3 modes of operation for moving bluefs files, determined by % of db used.
+   * %0
+   * A | move db_on_slow->db_on_db
+   * A | move slow_on_slow->slow_on_db
+   * %all_on_db
+   * B | move db_on_slow->db_on_db
+   * %no_slow_on_db
+   * C | move slow_on_db->slow_on_slow
+   * %100
+   *
+   * To prevent circular operation distance ( %all_on_db - %no_slow_on_db ) must be larger then largest file.
+   */
+  utime_t now = ceph_clock_now();
+  if (double(move_next_check - now) > 0) {
+    return;
+  }
+
+  std::unique_lock l(lock);
+  if (move_context)
```

Comment:
I can see that move_context is used as a boolean value so far - so may be better use atomic instead and do the check before the lock...